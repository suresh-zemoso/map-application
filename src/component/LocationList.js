import React, { useEffect, useState } from 'react'
import { useSelector, useDispatch } from "react-redux";
import { saveAll } from '../actions/actions';
import { locationService } from '../services/locationService';
import LocationTable from './LocationTable';



export default function LocationList() {

    const [locationList, setlocationList] = useState([]);
    const dispatch = useDispatch();
    const accessToken = useSelector(state => state.authentication.accessToken);

    useEffect(() => {
        locationService.getLocations(accessToken)
            .then(response => {
                const locations = response.data;
                setlocationList(locations);
                //save data in redux
                dispatch(saveAll(locations));
            })
            .catch(error => {
                console.log(error);
            })
    }, []);

    return (
        <LocationTable
            locationList={locationList}
            setlocationList={setlocationList}
        />
    )
}
