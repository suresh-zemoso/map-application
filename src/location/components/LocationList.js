import React, { useEffect, useState } from 'react'
import { useSelector, useDispatch } from "react-redux";
import { fetchLocations } from '../../location/actions/locationAction';
import LocationTable from './LocationTable';
import PropTypes from 'prop-types';



const defaultProps = {

}

const propTypes = {


}

const LocationList = (props) => {

    const { locations, loading, fetchError, errorMessage } = useSelector(state => state.locationState)
    const [locationList, setlocationList] = useState([]);

    const dispatch = useDispatch();
    const accessToken = useSelector(state => state.authentication.accessToken);


    useEffect(() => {
        setlocationList(locations)
    }, [locations])


    useEffect(() => {
        dispatch(fetchLocations(accessToken));
    }, []);

    return (
        <>
            {loading ? <div>Fetching Locations</div> : (fetchError ?
                <div>{errorMessage}</div> :
                <LocationTable
                    locationList={locationList}
                    setlocationList={setlocationList}
                />)
            }
        </>
    )
}

LocationList.propTypes = propTypes;
LocationList.defaultProps = defaultProps;

export default LocationList;