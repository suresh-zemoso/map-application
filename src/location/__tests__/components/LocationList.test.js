import React from 'react';
import { mount, shallow } from 'enzyme';
import LocationList from '../../components/LocationList';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';

const mockStore = configureMockStore([thunk]);
const store = mockStore({
    locationState: {
        locations: [], loading: false,
        fetchError: false, errorMessage: 'error in loading locations'
    }
});
let wrapped = shallow(<Provider store={store}><LocationList /></Provider>);

// const mockDispatch = jest.fn();

// jest.mock('react-redux', () => {
//     const state = {
//         locationState: {
//             locations: [], loading: false,
//             fetchError: false, errorMessage: 'error in loading locations'
//         }
//     };
//     return {
//         useSelector: jest.fn().mockImplementation(cb => cb(state)),
//         useDispatch: jest.fn().mockImplementation(() => mockDispatch),
//     };
// });

describe('LocationList', () => {

    it('should render the LocationList Component correctly', () => {
        expect(wrapped).toMatchSnapshot();
    });

});      