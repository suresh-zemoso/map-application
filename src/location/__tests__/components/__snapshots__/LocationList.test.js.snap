// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationList should render the LocationList Component correctly 1`] = `ShallowWrapper {}`;
